shepherd

```diff
@@ -73,61 +73,68 @@ update_services() {
 
   authenticate_to_registries
 
+  # iterate over all services
   for name in $(IFS=$'\n' docker service ls --quiet --filter "${FILTER_SERVICES}" --format '{{.Name}}'); do
+
+    # skip ignored services
+    if [[ "$(in_list "$ignorelist" "$name")" != "false" ]]; then
+      continue # continue with next service
+    fi
+
     local image_with_digest image auth_config config_flag
-    if [[ "$(in_list "$ignorelist" "$name")" == "false" ]]; then
-      image_with_digest="$(docker service inspect "$name" -f '{{.Spec.TaskTemplate.ContainerSpec.Image}}')"
-      image=$(echo "$image_with_digest" | cut -d@ -f1)
-      auth_config=$(docker service inspect "$name" -f '{{index .Spec.Labels "shepherd.auth.config"}}')
-      if [[ -z "$auth_config" ]]; then
-          config_flag=()
-      else
-          config_flag=(--config "$auth_config")
+
+    image_with_digest="$(docker service inspect "$name" -f '{{.Spec.TaskTemplate.ContainerSpec.Image}}')"
+    image=$(echo "$image_with_digest" | cut -d@ -f1)
+    auth_config=$(docker service inspect "$name" -f '{{index .Spec.Labels "shepherd.auth.config"}}')
+    if [[ -z "$auth_config" ]]; then
+      config_flag=()
+    else
+      config_flag=(--config "$auth_config")
+    fi
+
+    if ! DOCKER_CLI_EXPERIMENTAL=enabled docker "${config_flag[@]}" manifest inspect $insecure_registry_flag "$image" > /dev/null; then
+      logger "Error updating service $name! Image $image does not exist or it is not available"
+    else
+      logger "Trying to update service $name with image $image" "true"
+
+      # shellcheck disable=SC2086
+      if ! docker "${config_flag[@]}" service update "$name" $detach_option $registry_auth $no_resolve_image_flag ${UPDATE_OPTIONS} --image="$image" > /dev/null; then
+        logger "Service $name update failed on $hostname!"
+        if [[ "${ROLLBACK_ON_FAILURE+x}" ]]; then
+          logger "Rolling $name back"
+          # shellcheck disable=SC2086
+          docker "${config_flag[@]}" service update "$name" $detach_option $registry_auth $no_resolve_image_flag ${ROLLBACK_OPTIONS} --rollback > /dev/null
+        fi
+        if [[ "$apprise_sidecar_url" != "" ]]; then
+          title="[Shepherd] Service $name update failed on $hostname"
+          body="$(date) Service $name failed to update to $(docker service inspect "$name" -f '{{.Spec.TaskTemplate.ContainerSpec.Image}}')"
+          curl -X POST -H "Content-Type: application/json" --data "{\"title\": \"$title\", \"body\": \"$body\"}" "$apprise_sidecar_url"
+        fi
+        continue # continue with next service
       fi
 
-      if ! DOCKER_CLI_EXPERIMENTAL=enabled docker "${config_flag[@]}" manifest inspect $insecure_registry_flag "$image" > /dev/null; then
-        logger "Error updating service $name! Image $image does not exist or it is not available"
+      local previous_image, current_image
```

Comment:
Comma is not allowed here, it must be spaces only. (There is an error otherwise.)